imqsauth/http.go

```diff
@@ -468,17 +493,20 @@ func httpHandlerCreateUser(central *ImqsCentral, w http.ResponseWriter, r *httpR
 		password = authaus.RandomString(20, "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789")
 	}
 
-	if err := central.Central.CreateAuthenticatorIdentity(identity, password); err != nil {
+	if err := central.Central.CreateUserStoreIdentity(identity, username, firstname, lastname, mobilenumber, password); err != nil {
 		authaus.HttpSendTxt(w, http.StatusForbidden, err.Error())
 	} else {
-		if sendPasswordResetEmail {
-			code, msg := central.ResetPasswordStart(identity, true)
-			if code != http.StatusOK {
-				authaus.HttpSendTxt(w, http.StatusOK, fmt.Sprintf("Created identity '%v'. However, failed to initiate password reset: ", identity, msg))
-				return
+		// Get userId from identity just created
+		if userId, getUserIdErr := central.Central.GetUserIdFromIdentity(identity); getUserIdErr == nil {
```

Comment:
The `getUserIdErr != nil` path is not handled here. When writing HTTP handler functions, you need to perform manual "does this function always send a response" checking - the same as a compiler would verify for you if a function declares that it returns a value. It is illegal for an HTTP handler function not to send a response.